src/apps/chifra/internal/blocks/handle_count.go

```diff
@@ -18,8 +17,9 @@ import (
 )
 
 func (opts *BlocksOptions) HandleCounts() error {
-	rpcOptions := opts.Globals.DefaultRpcOptions(&globals.DefaultRpcOptionsSettings{
-		Opts: opts,
+	rpcOptions := rpcClient.DefaultRpcOptions(&rpcClient.DefaultRpcOptionsSettings{
```

Comment:
This is so excellent. Thanks so much for doing this...

I think one of the things this may imply is that all the RCP `Get` routines, which currently take both `chain` and `rpcOptions` could remove `chain` from their interface. Also, they could become more object oriented (which I like).

I'm not saying we should, but we could.

That would make

```
blocks, err := rpcClient.GetBlockByNumber(chain, bn, rpcOptions)
```

look like this:

```
blocks, err := rpcOptions.GetBlockByNumber(bn)
```

which, to me, is just SO much cleaner. We could even rename `rpcOptions` to something like `rpcConn` (connection) or `rpcProvider` or something.